jelly/src/main/java/org/kohsuke/stapler/jelly/ResourceBundle.java

```diff
@@ -153,14 +157,24 @@ protected Properties get(String key) {
         }
 
         if(in!=null) {
+            PropertyResourceBundle propertyResourceBundle;
             try {
-                try {
-                    props.load(in);
-                } finally {
-                    in.close();
-                }
+                propertyResourceBundle = new PropertyResourceBundle(in);
             } catch (IOException e) {
                 throw new Error("Failed to load "+url,e);
+            } finally {
+                try {
+                    in.close();
+                } catch (IOException ignored) {}
+            }
```

Comment:
This code could be simplified and the error handling improved to propagate the exception by removing these ancient (probably no longer necessary) workarounds and using a modern try-with-resources and `UncheckedIOException`:

```java
// attempt to load
props = new Properties();
String url = baseName + key + ".properties";
PropertyResourceBundle propertyResourceBundle;
try (InputStream in = openStream(url)) {
    propertyResourceBundle = new PropertyResourceBundle(in);
} catch (IOException e) {
    throw new UncheckedIOException("Failed to load " + url, e);
}
```

A mildly riskier refactoring, but I think it is probably safe. No strong feeling either way, though. Not a blocking comment.

Reply:
(If you don't want to go with the above refactoring, you could always call `e.addSuppressed(ignored)` to tape on the ignored exception from the `close()` to the primary exception.)

Reply:
Thanks, I had to make some minor adjustment as tests failed due to needing to skip missing locale specific files (due to our custom resource bundle like implementation :)

but I think this should be safe, will run it through a core PR to make sure though